### PR TITLE
Cleanup: removed dead/unnecessary classes

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
Removed dead/unnecessary classes:

 * VarargCapturingMatcher and Test
 * AnyVararg removed in Favor of Any, Any can now used as varargs matcher too 
 
 removed test [shouldNotAllowAnyObjectMatchEntireVararg()](https://github.com/mockito/mockito/blob/master/src/test/java/org/mockito/internal/invocation/ArgumentsComparatorTest.java#L110) cause it was not clear why ANY should not be allowed as varargs matcher

### CODE REVIEW

File: src/main/java/org/mockito/Matchers.java [hunk 1]
```diff
@@ -198,9 +198,12 @@ public static short anyShort() {
      * See examples in javadoc for {@link Matchers} class
      *
      * @return <code>null</code>.
+     * 
+     * @deprecated use {@link #any()}
```
Comment (TimvdLippe) [inline src/main/java/org/mockito/Matchers.java]: Per #482 please update this to `@deprecated as of 2.0, use {@link #any()}`

**Comment by TimvdLippe** [comment]
1 small documentation update, other than that LGTM

File: src/main/java/org/mockito/Matchers.java (hunk 1 above)
Comment (ChristianSchwarz) [inline src/main/java/org/mockito/Matchers.java]: fixed

**Comment by ChristianSchwarz** [comment]
@TimvdLippe fixed

**Pull request merged**